src/api_server/src/request/machine_configuration.rs

```diff
@@ -57,17 +63,24 @@ pub(crate) fn parse_patch_machine_config(body: &Body) -> Result<ParsedRequest, E
     )))
 }
 
-fn check_unsupported_fields(_vm_config: &VmConfig) -> Result<(), Error> {
-    #[cfg(target_arch = "aarch64")]
-    {
-        if _vm_config.cpu_template.is_some() {
-            // cpu_template is not supported on aarch64
-            return Err(Error::Generic(
-                StatusCode::BadRequest,
-                "CPU templates are not supported on aarch64".to_string(),
-            ));
-        }
+#[cfg(target_arch = "aarch64")]
+fn check_unsupported_fields(vm_config: &VmConfig) -> Result<(), Error> {
+    if vm_config.cpu_template.is_some() {
+        // cpu_template is not supported on aarch64
+        return Err(Error::Generic(
+            StatusCode::BadRequest,
+            "CPU templates are not supported on aarch64".to_string(),
+        ));
     }
+
+    if let Some(true) = vm_config.ht_enabled {
```

Comment:
I would disallow using this field on ARM altogether. What purpose does it serve to allow users to set it to false? It's just confusing.

```suggestion
    if vm_config.ht_enabled.is_some() {
```

If we go down this path, we could even have the `ht_enabled` field in the `VmConfig` struct marked with a `#[cfg(target_arch = "aarch64")]` so that we never mistakenly deserialize/serialize it.

Reply:
I generally agree, but I was trying to minimise the breaking change impact. Let's discuss this some more

Reply:
A breaking change is a breaking change is a breaking change.

Why do half the job to have only a breaking change in some cases? (setting `ht true on arm` will break, but `ht false on arm` will still work albeit it being useless/confusing)

Reply:
We've discussed this topic inside the team and reached the following path forward:

keep the behaviour of this PR but additionally rename the `ht_enabled` param to `smt` (it will be optional and return error if set to True on ARM). This enables us to add support for 'emulated' smt in ARM as well if we need to in the future. We think this is the most extensible and cleanest option.

WDYT @acatangiu ?

Reply:
I've implemented this in the latest commit

Reply:
Sounds reasonable, but I don't see why you're not disabling/removing it altogether on ARM. You could seamlessly add it later with default value `false` when/if you ever implement smt emulation for arm. And in the meantime you're not left with a stub optional param which allows only false or otherwise fails the request.

You're basically keeping a button saying "this button might be useful later, but don't push it now or it will error" :smile:

Reply:
> L.E.: an argument for keeping the stub is if that way the api requests would be identical between x86_64 and aarch64 -> more uniform code on customer side

This is one of the reasons I thought about.
It's also that otherwise, it may also be confusing for users to see that there is no ht_enabled config on arm. One may then wonder if it's emulated or not. The answer would be clearer if there was an ht_enabled option that defaults to `false` and that only allows False on Arm.